### ISSUE TITLE
Support for Pod Identity Add On for AKS

### ISSUE THREAD

**Issue by piizei** [hashicorp/terraform-provider-azurerm#9885]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please support the Pod Identity Add-On for AKS

### New or Affected Resource(s)

* azurerm_kubernetes_cluster

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster" "aks" {

  addon_profile {
    kube_dashboard {
      enabled = true
    }
    azure_policy {
      enabled = true
    }

    pod_identity {
      enabled = true
      principal_id = azurerm_user_assigned_identity.podid.principal_id
    }

  }

}
```

### References

* https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity

**Comment by Piotr-muzyka** [comment]
as this is fixed with #11492 can we please have it merged and included in the next release?

**Comment by avinashpancham** [comment]
Is there a timeline for Pod Identity on AKS being supported by Terraform? I understand that Azure will release a V2 of Pod Identity but that will only be available in [Q2 of 2022](https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity), which is between 6 and 9 months away. 

Having a recommend temporary method to turn this on would be nice, even if it just would be as shown below ( I dont know if this works, but this is what I would try at first)
```bash
resource "azurerm_kubernetes_cluster" "default" {
  name                   = "my_cluster"
  location               = "westeurope"

  provisioner "local-exec" {
    command = "az aks update -g my_resource_group -n my_cluster --enable-pod-identity"
  }

**Comment by avinashpancham** [comment]
For anyone struggling with this: I found that Azure also shares these functionalities as [Helm charts](https://azure.github.io/aad-pod-identity/docs/getting-started/installation/). So you can just create a regular AKS cluster with Terraform and instead of enabling the pod-identity by adding the `local-exec` as above, you apply the following Helm chart with the kubernetes Helm provider:

```
resource "helm_release" "aad-pod-identity" {
  name       = "aad-pod-identity"
  repository = "https://raw.githubusercontent.com/Azure/aad-pod-identity/master/charts"
  chart      = "aad-pod-identity"
  version    = "4.1.5"  # use the latest version at the time of reading
}
```

Azure offers these Helm charts for multiple preview features that are not yet supported by the AKS resource in the azurerm provider. Next to Azure Pod Identity, a Helm chart is for example also available for the AKS [Secrets Store CSI Driver](https://azure.github.io/secrets-store-csi-driver-provider-azure/getting-started/installation/)

**Comment by vipulp86** [comment]
Currently, POD identity(with its k8s resources) add-on on cluster can be enabled using Azure Preview [feature](https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity).

But we need a terraform azure RM resource for AKS Pod Identity to be assigned to kubernetes namespaces. As of now it gets assigned via [azure CLI](https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity#create-a-pod-identity) to individual namespaces.

**Comment by katbyte** [comment]
@vipulp86 - that is for the older v1 pod identity feature. as mentioned in the linked documentation it will be replaced by the new [Azure AD Workload Identity](https://github.com/Azure/AKS/issues/1480) api which is schedule for public preview sometime this [month](https://github.com/Azure/AKS/issues/1480#issuecomment-1016373110) - once thats out we intend to  look into implementing it.

**Comment by Vegoo89** [comment]
I follow the topic regarding Workload Identity and it seems that support for Managed Identity won't be added before 9/30.
If someone implements support for V1 addon, would it be possible to merge the changes or you don't want to touch V1 at all?

Right now - for me - to make it work I have to call Azure management API to update cluster to use the addon along with identities and exceptions, which is far from ideal solution.

I saw pretty good merge request that was closed already and just wondering :)
